ganga/GangaCore/Core/MonitoringComponent/Local_GangaMC_Service.py

```diff
@@ -636,6 +650,18 @@ def __monStep(self):
         self.__updateTimeStamp = time.time()
         self.__sleepCounter = config['base_poll_rate']
 
+    
+    def reloadJob(self, i):
+        """
+        Reload a Job from disk.
+        Parameters:
+        i: Job ID
+        Return:
+            True, if Job is successfully reloaded from disk.
+        """
+        stripProxy(self.registry_slice).objects.repository.load([i])
```

Comment:
I think from code inspection `registry_slice` is a `RegistrySlice` object. Therefore the `objects` attribute is an oDict. Shouldn't you be calling the first entry on the oDict here?

Also this is then an object of type `Job` which doesn't have a repository attribute or have I missed what is being called here?

I think what you're after is something similar to:
`jobs.select(47,47)._impl.objects.values()[0]._registry.repository.load([47])`

Reply:
Yes, registry_slice is a RegistrySlice object but `objects` is `JobRegistry` object.
I'm accessing `load` from `GangaRepositoryXML` in similar way as `update_index` method is being called here:

https://github.com/apsknight/ganga/blob/95b6de6aff10947cb4a56ad1d9d2877ab08b39c3/ganga/GangaCore/Core/MonitoringComponent/Local_GangaMC_Service.py#L550

Is there any better way to achieve this reloading of Job from disk. I tried modifying the getter here: 
https://github.com/ganga-devs/ganga/blob/develop/ganga/GangaCore/GPIDev/Base/Objects.py#L328 

but couldn't make it work.

Reply:
OK, the `RegistrySlice` object initially passed has a different form to the object than is constructed from `jobs.select` this is bad behavior and I suspect to fix that behavior requires some serious work in the XXXRegistry classes... (not for now)

I suspect this was an attempt to get the monitoring to pick up new jobs through a dynamically updating reference list but this seems to cause more problems probably due to the confusing different ways `RegistrySlice` can be constructed. This was likely done before the proxy layer was introduced and the new proxy objects which can be constructed at runtime are static objects which have an implicitly different form to the dynamic one.


Exposing the `reloadJob`method will almost certainly fail if called with something like `reloadJob(jobs.select(a,a))` as the object will in this instance be an oDict so it will either need refactoring or at least not exposing through the GPI as `RegistrySlice` objects don't use a very standardized interface. I'd at least suggest not exposing it for now unless this is needed, in which case some work will need to be done to consider the case of `reloadJob(jobs.select(1,1))` from the GPI.


Aside from clearing the contents of an object before re-loading it from disk there is no better fix for things getting out of sync. 


I did some digging and I've spotted at least 1 bug in the monitoring code.
running `runMonitoring(jobs.select(a,b))` will cause all future monitoring to not work correctly as the dynamic job list is incorrectly replaced by a static list of just the objects in the select... this is potentially annoying and might explain some problems I've seen in the past... I might suggest a short PR to fix this.

Reply:
@rob-c @apsknight what is the status with this?

Reply:
@mesmith75 I think removing this method from the GPI, or adding some inspection for the slice being passed through should be enough, depending on what is needed. I'm tempted to say this _should_ go in, but it won't work as expected for people using `runMonitoring` given a bug in the monitoring code.

As for a bug in the monitoring: https://github.com/ganga-devs/ganga/blob/develop/ganga/GangaCore/Core/MonitoringComponent/Local_GangaMC_Service.py#L699 appears to be incorrect as it will potentially replace the main jobresgistry object with a static oDict behind the scenes.

I would suggest a fix but which should just be to remove this line and pass through a custom slice to the methods in the monitoring. This bug exposes itself such that `runMonitoring` works fine but attempting to enable `enableMonitoring` after this will cause it to fault. As would running `runMonitoring(...)` with the core monitoring task enabled as this will replace the monitoring loop with only the jobs requested by the `runMonitoring` method which isn't the correct behavior.
Apologies I've not had the time to make a PR over this I'm a bit busy and only just spotted the bug in the logic after re-visiting it recently, so if someone else wants to step up it should be an easy fix.

I've no idea if the problem with re-loading subjobs remains but that I think is a separate issue to this PR.